module/signature/signer_store.go

```diff
@@ -0,0 +1,64 @@
+package signature
+
+import (
+	"github.com/onflow/flow-go/model/encoding"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/storage"
+)
+
+// EpochAwareSignerStore implements the SignerStore interface. It is epoch
+// aware, and provides the appropriate threshold signers on a per-view basis,
+// using the database to retrieve the relevant DKG keys.
+type EpochAwareSignerStore struct {
+	epochSetups storage.EpochSetups               // used to fetch epoch counter by view
+	keys        storage.DKGKeys                   // used to fetch DKG private key by epoch
+	signers     map[uint64]module.ThresholdSigner // cache of signers by epoch
+}
+
+// NewEpochAwareSignerStore instantiates a new EpochAwareSignerStore
+func NewEpochAwareSignerStore(epochSetups storage.EpochSetups, keys storage.DKGKeys) *EpochAwareSignerStore {
+	return &EpochAwareSignerStore{
+		epochSetups: epochSetups,
+		keys:        keys,
+		signers:     make(map[uint64]module.ThresholdSigner),
+	}
+}
+
+// GetSigner returns the threshold-signer for signing objects at a given view.
+// The view determines the epoch, which determines the DKG private key
+// underlying the signer.
+func (s *EpochAwareSignerStore) GetSigner(view uint64) (module.ThresholdSigner, error) {
+	epoch, err := s.epochSetups.CounterByView(view)
+	if err != nil {
+		return nil, err
+	}
+	sig, ok := s.signers[epoch]
```

Comment:
```suggestion
	signer, ok := s.signers[epoch]
```
Might be a bit clearer